### PR TITLE
feat: Add synthetic for mechanism

### PR DESCRIPTION
**Pull request by philipphofmann**
## :scroll: Description

Add synthetic property for SentryMechanism as it's part of the protocol https://develop.sentry.dev/sdk/event-payloads/exception/#exception-mechanism

## :bulb: Motivation and Context

https://github.com/getsentry/sentry-cocoa/issues/1192

## :green_heart: How did you test it?
Unit tests

## :pencil: Checklist

- [x] I reviewed the submitted code
- [x] I added tests to verify the changes
- [x] I updated the docs if needed
- [x] Review from the native team if needed
- [x] No breaking changes

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,7 @@ This version adds a dependency on Swift.
 
 - Properly demangle Swift class name (#2162)
 - Enable File I/O APM by default (#2497)
+- feat: Add synthetic for mechanism (#2501)
```
Comment (kevinrenskers) [inline CHANGELOG.md:11]:
```suggestion
- Add synthetic for mechanism (#2501)
```

File: Sources/Sentry/Public/SentryMechanism.h [hunk 2]
```diff
@@ -35,6 +35,12 @@ SENTRY_NO_INIT
  */
 @property (nonatomic, copy) NSNumber *_Nullable handled;
 
+/**
+ * An optional flag indicating a synthetic exception. For more info visit
+ * https://develop.sentry.dev/sdk/event-payloads/exception/#exception-mechanism.
+ */
+@property (nonatomic, copy, nullable) NSNumber *synthetic;
```
Comment (kevinrenskers) [inline Sources/Sentry/Public/SentryMechanism.h:42]: I think this should be simply a `BOOL`, as that's what this flag is in the end.

Comment (brustolin) [reply Sources/Sentry/Public/SentryMechanism.h:42]: I understand that we're using NSNumber to have a nullable boolean, but is this really necessary? 
Is it "undefined" really an option?

**Review by brustolin — APPROVED** [review]
LGTM

File: Sources/Sentry/Public/SentryMechanism.h (hunk 2 above)
Comment (philipphofmann) [reply Sources/Sentry/Public/SentryMechanism.h:42]: Yes, it's undefined. We don't send it currently. I would like to keep it, so the payload is smaller.

**Pull request merged**